plugins/cros_ec-usb/fu-cros_ec-usb-device.c

```diff
@@ -0,0 +1,309 @@
+	 * once we figure out what's going on.
+	GUsbDevice *usb_device = fu_usb_device_get_dev (FU_USB_DEVICE (device));
+	guint8 inbuf[64];
+	gsize actual = 0;
+	while (g_usb_device_bulk_transfer (usb_device,
+					   self->ep_num | 0x80,
+					   inbuf, self->chunk_len,
+					   &actual, 10, NULL, error)) {
+
+		g_debug ("flush!\n");
+	}
+	*/
+
+	memset(&ufh, 0, sizeof(ufh));
+	ufh.block_size = htobe32(sizeof(ufh));
+	if (!do_xfer (self, (guint8 *)&ufh, sizeof(ufh), (guint8 *)&start_resp,
+		      sizeof(start_resp), TRUE, &rxed_size, error))
+		return FALSE;
+
+	/* We got something. Check for errors in response */
+	if (rxed_size < 8) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_PARTIAL_INPUT,
+			     "Unexpected response size %" G_GSIZE_FORMAT ": \n",
+			     rxed_size);
+		return FALSE;
+	}
+	self->protocol_version = (guint16)be16toh (start_resp.rpdu.protocol_version);
+
+	if (self->protocol_version < 5 || self->protocol_version > 6) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_NOT_SUPPORTED,
+			     "Unsupported protocol version %d: \n",
+			     self->protocol_version);
+		return FALSE;
+	}
+	self->header_type = (guint16)be16toh (start_resp.rpdu.header_type);
+
+	g_debug ("target running protocol version %d (type %d)\n",
+		 self->protocol_version, self->header_type);
+
+	error_code = (guint32)be32toh (start_resp.rpdu.return_value);
+	if (error_code) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_FAILED,
+			     "Target reporting error %u: \n", error_code);
+		return FALSE;
+	}
+
+	memcpy(self->targ.common.version, start_resp.rpdu.common.version,
+		sizeof(start_resp.rpdu.common.version));
+	self->targ.common.maximum_pdu_size =
+		be32toh (start_resp.rpdu.common.maximum_pdu_size);
+	self->targ.common.flash_protection =
+		be32toh (start_resp.rpdu.common.flash_protection);
+	self->targ.common.min_rollback = be32toh (start_resp.rpdu.common.min_rollback);
+	self->targ.common.key_version = be32toh (start_resp.rpdu.common.key_version);
+
+	g_debug ("maximum PDU size: %u\n", self->targ.common.maximum_pdu_size);
+	g_debug ("Flash protection status: %04x\n", self->targ.common.flash_protection);
+	g_debug ("version: %32s\n", self->targ.common.version);
+	g_debug ("key_version: %u\n", self->targ.common.key_version);
+	g_debug ("min_rollback: %d\n", self->targ.common.min_rollback);
+
+	if (!fu_cros_ec_usb_parse_version (self->targ.common.version,
+					   &self->version)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "Unable to parse version: %32s\n", self->targ.common.version);
+		return FALSE;
+	}
+
+	g_debug ("boardname = %s\n", self->version.boardname);
+	g_debug ("triplet = %s\n", self->version.triplet);
+	g_debug ("sha1 = %s\n", self->version.sha1);
+	g_debug ("%s\n", self->version.dirty?"dirty":"clean");
+
+	fu_device_set_version (FU_DEVICE(device), self->version.triplet);
+	fu_device_add_guid (FU_DEVICE(device), self->version.boardname);
```

Comment:
I think that you actually want to use `fu_device_add_instance_id` usually here instead of `fu_device_add_guid`.  IIRC this allows quirks to be applied (if they end up being necessary).

Reply:
Done.